src/main/java/fr/inria/coming/main/ComingMain.java

```diff
@@ -59,35 +59,69 @@
+                .argName("path").hasArg()
+                .desc("analyse the content in \'path\'")
+                .build());
+
+        options.addOption(Option.builder("mode")
+                .argName("mineinstance | diff | features").hasArg()
+                .desc("the mode of execution of the analysis")
+                .build());
+
+        options.addOption(Option.builder("input")
+                .argName("git(default) | files").hasArg()
+                .desc("format of the content present in the given -path. git implies that the path is a git repository. files implies the path contains .patch files ")
+                .build());
+
+        options.addOption(Option.builder("output")
+                .argName("path").hasArg()
+                .desc("dump the output of the analysis in the given path")
+                .build());
+
+        options.addOption(Option.builder("outputprocessor")
+                .argName("classname").hasArg()
+                .desc("output processors for result")
+                .build()); // TODO: What's the default and what is changefrequency. And oh btw it's a list.
+
+        // Pattern mining
+        options.addOption(Option.builder("pattern")
+                .argName("path").hasArg()
+                .desc("path of the pattern file to be used when the -mode is \'mineinstance\'")
+                .build());
+
+        options.addOption(Option.builder("patternparser")
+                .argName("classname").hasArg()
+                .desc("parser to be used for parsing the file specified -pattern. Default is XML")
+                .build());
+
+        options.addOption("entitytype", true, "entity type to be mine");
+        options.addOption("entityvalue", true, "the value of the entity  mentioned in -entitytype");
+
+        options.addOption(Option.builder("action")
+                .argName("INS | DEL | UPD | MOV | PER | ANY").hasArg()
+                .desc("tye of action to be mined")
+                .build());
+
+        options.addOption("parenttype", true, "parent type of the nodes to be considered");
+        options.addOption("parentlevel", true, "numbers of AST node where the parent is located. 1 implies immediate parent");
+
+        options.addOption("hunkanalysis", true, "include analysis of hunks");
+
+        options.addOption("showactions", false, "show all actions");
+        options.addOption("showentities", false, "show all entities");
+
+        // Revision filter
+        options.addOption("filter", true, "name of the filter");
+        options.addOption("filtervalue", true, "values of the filter  mentioned in -filter");
+
+        // In case of git
+        options.addOption(Option.builder("branch")
+                .argName("branch name").hasArg()
+                .desc("In case of -input=\'git\', use this branch name. Default is master.")
+                .build());
+        options.addOption("message", true, "comming message"); // TODO: What's this?
```

Comment:
What is the purpose of this argument?

Reply:
> What is the purpose of this argument?

Using this argument you can specify the branch to analyze. By default Coming analyzes the branch Master

Reply:
I was talking about the message argument.